project.php

```diff
@@ -1974,6 +1969,120 @@ function sr_echo_time_form($label, $min_days, $max_days, $default_days, $extend
     echo "</form>\n";
 }
 
+function echo_smoothreading_options($project)
+{
+    global $projects_url;
+
+    $smooth_dir = "$project->dir/smooth";
+    $project_url = "$projects_url/$project->projectid";
+    $smooth_url = "$project_url/smooth";
+    echo "<li class='list-head'>", _("Download a Smooth Reading file");
+    echo "<ul>";
+    echo_file_downloads(glob("$smooth_dir/*.txt"), $smooth_url);
+
+    // download zipped html
+    $html_files = glob("$smooth_dir/*.{htm,html}", GLOB_BRACE);
+    if($html_files)
+    {
+        // assume only one html file
+        $file = $html_files[0];
+        $file_base_name = basename($file);
+        // zip the html file with images folder for download
+        $zip = new ZipArchive;
```

Comment:
This will recreate the .zip archive on every page load, which we shouldn't be doing. We should create the .zip file of the HTML on upload and only display it on page load.

Reply:
Good point.